drivers/gpio/gpio_tle9104.c

```diff
@@ -0,0 +1,544 @@
+		LOG_ERR("no power on reset detected");
+		return -EIO;
+	}
+
+	result = tle9104_write_register(dev, TLE9104REGISTER_CFG, register_cfg);
+	if (result != 0) {
+		LOG_ERR("unable to write configuration");
+		return result;
+	}
+
+	register_globalstatus = 0x00;
+	/* disable communication watchdog */
+	tle9104_set_cfg_cwdtime(&register_cfg, 0);
+	/* enable outputs */
+	register_globalstatus |= TLE9104_GLOBALSTATUS_OUTEN_BIT;
+
+	result = tle9104_write_register(dev, TLE9104REGISTER_GLOBALSTATUS, register_globalstatus);
+	if (result != 0) {
+		LOG_ERR("unable to write global status");
+		return result;
+	}
+
+	return 0;
+}
+
+BUILD_ASSERT(CONFIG_GPIO_TLE9104_INIT_PRIORITY > CONFIG_SPI_INIT_PRIORITY,
+	     "TLE9104 must be initialized after SPI");
+
+#define TLE9104_INIT_GPIO_FIELDS(inst, gpio)                                                       \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, gpio),                                             \
+		    (GPIO_DT_SPEC_GET_BY_IDX(DT_DRV_INST(inst), gpio, 0)), ({0}))
+
+#define TLE9104_INIT(inst)                                                                         \
+	static const struct tle9104_config tle9104_##inst##_config = {                             \
+		.common =                                                                          \
+			{                                                                          \
+				.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(inst),            \
+			},                                                                         \
+		.bus = SPI_DT_SPEC_INST_GET(                                                       \
+			inst, SPI_OP_MODE_MASTER | SPI_MODE_CPHA | SPI_WORD_SET(8), 0),            \
+		.gpio_enable = TLE9104_INIT_GPIO_FIELDS(inst, en_gpios),                           \
+		.gpio_reset = GPIO_DT_SPEC_GET_BY_IDX(DT_DRV_INST(inst), resn_gpios, 0),           \
+		.gpio_control =                                                                    \
+			{                                                                          \
+				TLE9104_INIT_GPIO_FIELDS(inst, in1_gpios),                         \
+				TLE9104_INIT_GPIO_FIELDS(inst, in2_gpios),                         \
+				TLE9104_INIT_GPIO_FIELDS(inst, in3_gpios),                         \
+				TLE9104_INIT_GPIO_FIELDS(inst, in4_gpios),                         \
+			},                                                                         \
+	};                                                                                         \
+                                                                                                   \
+	static struct tle9104_data tle9104_##inst##_drvdata = {                                    \
+		.state = 0x00,                                                                     \
+		.configured = 0x00,                                                                \
+	};                                                                                         \
```

Comment:
no need to initialize this, static data is zero initialized automatically

Reply:
done